File: test-suite-python/src/test/python/micronaut/docs/client/ApiClientListener.py
from java.util.concurrent import TimeUnit
from jakarta.inject import Singleton
from micronaut.context.event import BeanCreatedEvent, BeanCreatedEventListener

from io.kubernetes.client.openapi import ApiClient


# tag::listener[]
@Singleton
class ApiClientListener(BeanCreatedEventListener[ApiClient]):

    def onCreated(self, event: BeanCreatedEvent[ApiClient]) -> ApiClient:
        api_client = event.getBean()
        ok_http_client = api_client.getHttpClient().newBuilder() \
            .readTimeout(5345, TimeUnit.MILLISECONDS) \
            .build()
        api_client.setHttpClient(ok_http_client)
        return api_client
# end::listener[]

File: test-suite/src/test/java/io/micronaut/docs/client/ApiClientListener.java
package io.micronaut.docs.client;

import io.kubernetes.client.openapi.ApiClient;
import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import jakarta.inject.Singleton;
import okhttp3.OkHttpClient;

import java.util.concurrent.TimeUnit;

//tag::listener[]
@Singleton
public class ApiClientListener implements BeanCreatedEventListener<ApiClient> {

    @Override
    public ApiClient onCreated(BeanCreatedEvent<ApiClient> event) {
        ApiClient apiClient = event.getBean();
        OkHttpClient okHttpClient = apiClient.getHttpClient().newBuilder()
                .readTimeout(5345, TimeUnit.MILLISECONDS)
                .build();
        apiClient.setHttpClient(okHttpClient);
        return apiClient;
    }
}
//end::listener[]
